src/main/thrift/parquet.thrift

```diff
@@ -737,6 +755,66 @@ union ColumnOrder {
   1: TypeDefinedOrder TYPE_ORDER;
 }
 
+struct PageLocation {
+  /** Offset of the page in the file **/
+  1: required i64 offset
+
+  /**
+   * Size of the page, including header. Sum of compressed_page_size and header
+   * length
+   */
+  2: required i32 compressed_page_size
+
+  /**
+   * Index within the RowGroup of the first row of the page; this means pages
+   * change on record boundaries (r = 0).
+   */
+  3: required i64 first_row_index
+}
+
+struct OffsetIndex {
+  /**
+   * PageLocations, ordered by increasing PageLocation.offset. It is required
+   * that page_locations[i].first_row_index < page_locations[i+1].first_row_index.
+   */
+  1: required list<PageLocation> page_locations
+}
+
+/**
+ * Description for ColumnIndex.
+ * Each <array-field>[i] refers to the page at OffsetIndex.page_locations[i]
+ */
+struct ColumnIndex {
+  /**
+   * A list of Boolean values to determine the validity of the corresponding
+   * min and max values. If true, a page contains only null values, and writers
+   * have to set the corresponding entries in min_values and max_values to
+   * byte[0], so that all lists have the same length. If false, the
+   * corresponding entries in min_values and max_values must be valid.
+   */
+  1: required list<bool> null_pages
+
+  /**
+   * Two lists containing lower and upper bounds for the values of each page.
+   * These may be the actual minimum and maximum values found on a page, but can
+   * also be (more compact) values that does not exist on a page. Readers must
+   * make sure that values are valid before using them by inspecting null_pages.
+   */
+  2: required list<binary> min_values
+  3: required list<binary> max_values
+
+  /**
+   * Stores whether both min_values and max_values are orderd and if so, in
+   * which order. This allows readers to perform binary searches in both lists.
+   * Readers cannot assume that max_values[i] <= min_values[i+1], even if the
```

Comment:
I agree with this design decision and understand the reason behind it, but it may be hard for readers to understand why this is good (actual_max[i]="Blart Versenwald III", actual_min[i+1]="Blart Versenwald IV", stored_max[i]="C", stored_min[i+1]="B"). I would suggest elaborating a bit on this in the .md file and maybe also putting a note in the comment of this struct to refer to the documentation for details.

Reply:
Do you have a suggestion what to add so it becomes more clear?

Reply:
Is this so you can truncate very large values (e.g., strings)? I think it would be helpful to point that out in the .md (ie, something along the lines that the writer can/should enforce some reasonable bound on the size of the index structures).